rclcpp/test/rclcpp/test_type_support.cpp

```diff
@@ -0,0 +1,201 @@
+const rosidl_message_type_support_t * ts_parameter_event =
+  rclcpp::type_support::get_parameter_event_msg_type_support();
+
+class TestTSParameterEvent : public rclcpp::PublisherBase
+{
+public:
+  explicit TestTSParameterEvent(rclcpp::Node * node)
+  : rclcpp::PublisherBase(
+      node->get_node_base_interface().get(),
+      "topicTSParameterEvent",
+      *ts_parameter_event,
+      PublisherOptions()) {}
+};
+
+const rosidl_message_type_support_t * ts_set_parameter_result =
+  rclcpp::type_support::get_set_parameters_result_msg_type_support();
+
+class TestTSSetParameterResult : public rclcpp::PublisherBase
+{
+public:
+  explicit TestTSSetParameterResult(rclcpp::Node * node)
+  : rclcpp::PublisherBase(
+      node->get_node_base_interface().get(),
+      "topicTSSetParameterResult",
+      *ts_set_parameter_result,
+      PublisherOptions()) {}
+};
+
+const rosidl_message_type_support_t * ts_parameter_descriptor =
+  rclcpp::type_support::get_parameter_descriptor_msg_type_support();
+
+class TestTSParameterDescriptor : public rclcpp::PublisherBase
+{
+public:
+  explicit TestTSParameterDescriptor(rclcpp::Node * node)
+  : rclcpp::PublisherBase(
+      node->get_node_base_interface().get(),
+      "topicTSParameterDescriptor",
+      *ts_parameter_descriptor,
+      PublisherOptions()) {}
+};
+
+const rosidl_message_type_support_t * ts_list_parameter_result =
+  rclcpp::type_support::get_list_parameters_result_msg_type_support();
+
+class TestTSListParametersResult : public rclcpp::PublisherBase
+{
+public:
+  explicit TestTSListParametersResult(rclcpp::Node * node)
+  : rclcpp::PublisherBase(
+      node->get_node_base_interface().get(),
+      "topicTSListParametersResult",
+      *ts_list_parameter_result,
+      PublisherOptions()) {}
+};
+
+/*
+   Test that the publisher is created properly for different msg typesupport
+ */
+TEST_F(TestTypeSupport, basic_getters) {
+  {
+    auto publisher = TestTSParameterEvent(node.get());
+    std::shared_ptr<const rcl_publisher_t> publisher_handle = publisher.get_publisher_handle();
+    EXPECT_NE(nullptr, publisher_handle);
+  }
+  {
+    auto publisher = TestTSSetParameterResult(node.get());
+    std::shared_ptr<const rcl_publisher_t> publisher_handle = publisher.get_publisher_handle();
+    EXPECT_NE(nullptr, publisher_handle);
+  }
+  {
+    auto publisher = TestTSParameterDescriptor(node.get());
+    std::shared_ptr<const rcl_publisher_t> publisher_handle = publisher.get_publisher_handle();
+    EXPECT_NE(nullptr, publisher_handle);
+  }
+  {
+    auto publisher = TestTSListParametersResult(node.get());
+    std::shared_ptr<const rcl_publisher_t> publisher_handle = publisher.get_publisher_handle();
+    EXPECT_NE(nullptr, publisher_handle);
+  }
+}
+
+/* Testing type support getters */
+TEST_F(TestTypeSupport, test_service_ts_get_params_srv) {
+  const rosidl_service_type_support_t * ts =
+    rclcpp::type_support::get_get_parameters_srv_type_support();
+  EXPECT_NE(nullptr, ts);
```

Comment:
I was sort of mistaken earlier. I think ASSERT_NE makes more sense here since the pointer is needed for `test_type_support_init_fini`

Reply:
Agree, see 5b0137c